src/service-override/extensionGallery.ts

```diff
@@ -0,0 +1,72 @@
+import { ExtensionManagementService } from 'vs/workbench/services/extensionManagement/common/extensionManagementService'
+import { WebExtensionManagementService } from 'vs/workbench/services/extensionManagement/common/webExtensionManagementService'
+import 'vs/workbench/contrib/extensions/browser/extensions.contribution'
+import 'vs/workbench/contrib/extensions/browser/extensions.web.contribution'
+import 'vs/workbench/contrib/logs/common/logs.contribution'
+import { IRemoteUserDataProfilesService, RemoteUserDataProfilesService } from 'vs/workbench/services/userDataProfile/common/remoteUserDataProfiles'
+import { ExtensionEnablementService } from 'vs/workbench/services/extensionManagement/browser/extensionEnablementService'
+import 'vs/workbench/services/extensionManagement/browser/extensionBisect'
+import { changeUrlDomain } from './tools/url'
+import { registerAssets } from '../assets'
+
+registerAssets({
+  'vs/workbench/services/extensions/worker/webWorkerExtensionHostIframe.html': () => changeUrlDomain(new URL('../assets/webWorkerExtensionHostIframe.html', import.meta.url).href, undefined)
+})
+
+export interface ExtensionGalleryOptions {
+  /**
+   * Whether we should only allow for web extensions to be installed, this is generally
+   * true if there is no server part.
+   */
+  webOnly: boolean
+}
+
+export default function getServiceOverride (options: ExtensionGalleryOptions = { webOnly: false }): IEditorOverrideServices {
+  return {
+    [ILocaleService.toString()]: new SyncDescriptor(WebLocaleService, [], false),
+    [IExtensionGalleryService.toString()]: new SyncDescriptor(ExtensionGalleryService, [], true),
+    [IGlobalExtensionEnablementService.toString()]: new SyncDescriptor(GlobalExtensionEnablementService, [], true),
+    [IExtensionsWorkbenchService.toString()]: new SyncDescriptor(ExtensionsWorkbenchService, [], true),
+    [IExtensionManagementServerService.toString()]: new SyncDescriptor(ExtensionManagementServerService, [], true),
+    [IExtensionRecommendationsService.toString()]: new SyncDescriptor(ExtensionRecommendationsService, [], true),
+    [IExtensionRecommendationNotificationService.toString()]: new SyncDescriptor(ExtensionRecommendationNotificationService, [], true),
+    [IWebExtensionsScannerService.toString()]: new SyncDescriptor(WebExtensionsScannerService, [], true),
+    [IExtensionIgnoredRecommendationsService.toString()]: new SyncDescriptor(ExtensionIgnoredRecommendationsService, [], true),
+    [IIgnoredExtensionsManagementService.toString()]: new SyncDescriptor(IgnoredExtensionsManagementService, [], true),
+    [IExtensionManifestPropertiesService.toString()]: new SyncDescriptor(ExtensionManifestPropertiesService, [], true),
+    [IExtensionManagementService.toString()]: options.webOnly
+      ? new SyncDescriptor(WebExtensionManagementService, [], true)
+      : new SyncDescriptor(ExtensionManagementService, [], true),
```

Comment:
I'm a little lost here, it seems in vscode, `WebExtensionManagementService` seems to only be used inside `ExtensionManagementServerService`

`IExtensionManagementService` (via `IWorkbenchExtensionManagementService`) seems to only be bound to `ExtensionManagementService` in vscode web

is it a `hack` to disable running extension on the server when the server connection is available?

Reply:
I do think it's an interesting feature, I'm just not sure about the current implementation. Is there an issue with my suggestion?

Reply:
Added this!